Formatting code is giving exceptions in Intellij on Windows

When I try to format my camel code in Intellij it is not providing the correct format.
On my linux machine it is working fine but not on my windows machine.

this is the stacktrace: `Got unexpected exception during formatting file://MyClass.java

java.util.concurrent.ExecutionException: java.lang.IndexOutOfBoundsException: chars sequence.length:14571, start:13620, end:16657
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$ProcessingTask.checkStop(AbstractLayoutCodeProcessor.java:483)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$ProcessingTask.performFileProcessing(AbstractLayoutCodeProcessor.java:477)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor.lambda$runProcessFile$0(AbstractLayoutCodeProcessor.java:293)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$2.run(AbstractLayoutCodeProcessor.java:316)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:429)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:114)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressAsynchronously$6(CoreProgressManager.java:480)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$3(ProgressRunner.java:252)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:186)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:173)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:252)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IndexOutOfBoundsException: chars sequence.length:14571, start:13620, end:16657
	at com.intellij.util.text.CharSequenceSubSequence.<init>(CharSequenceSubSequence.java:20)
	at com.intellij.util.text.ImmutableText.subSequence(ImmutableText.java:206)
	at com.intellij.openapi.editor.impl.DocumentImpl$1.subSequence(DocumentImpl.java:93)
	at com.github.cameltooling.idea.service.extension.camel.JavaCamelIdeaUtils.formatText(JavaCamelIdeaUtils.java:478)
	at com.github.cameltooling.idea.service.extension.camel.JavaCamelIdeaUtils.lambda$format$4(JavaCamelIdeaUtils.java:442)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:958)
	at com.github.cameltooling.idea.service.extension.camel.JavaCamelIdeaUtils.format(JavaCamelIdeaUtils.java:442)
	at com.github.cameltooling.idea.service.extension.camel.JavaCamelIdeaUtils.processText(JavaCamelIdeaUtils.java:419)
	at com.github.cameltooling.idea.util.CamelIdeaUtils.processText(CamelIdeaUtils.java:285)
	at com.github.cameltooling.idea.formatter.CamelPostFormatProcessor.processText(CamelPostFormatProcessor.java:41)
	at com.intellij.psi.impl.source.codeStyle.CoreCodeStyleUtil.postProcessRange(CoreCodeStyleUtil.java:147)
	at com.intellij.psi.impl.source.codeStyle.CoreCodeStyleUtil.postProcessEnabledRanges(CoreCodeStyleUtil.java:138)
	at com.intellij.psi.impl.source.codeStyle.CoreCodeStyleUtil.postProcessText(CoreCodeStyleUtil.java:126)
	at com.intellij.formatting.service.CoreFormattingService.lambda$formatRanges$0(CoreFormattingService.java:64)
	at com.intellij.psi.impl.source.codeStyle.CoreCodeStyleUtil.postProcessRanges(CoreCodeStyleUtil.java:114)
	at com.intellij.formatting.service.CoreFormattingService.formatRanges(CoreFormattingService.java:64)
	at com.intellij.formatting.service.FormattingServiceUtil.formatRanges(FormattingServiceUtil.java:93)
	at com.intellij.psi.impl.source.codeStyle.CodeStyleManagerImpl.reformatText(CodeStyleManagerImpl.java:151)
	at com.intellij.psi.impl.source.codeStyle.CodeStyleManagerImpl.reformatText(CodeStyleManagerImpl.java:114)
	at com.intellij.codeInsight.actions.ReformatCodeProcessor.lambda$doReformat$5(ReformatCodeProcessor.java:201)
	at com.intellij.util.SlowOperations.allowSlowOperations(SlowOperations.java:194)
	at com.intellij.codeInsight.actions.ReformatCodeProcessor.lambda$doReformat$6(ReformatCodeProcessor.java:185)
	at com.intellij.openapi.editor.ex.util.EditorScrollingPositionKeeper.perform(EditorScrollingPositionKeeper.java:101)
	at com.intellij.codeInsight.actions.ReformatCodeProcessor.doReformat(ReformatCodeProcessor.java:185)
	at com.intellij.codeInsight.actions.ReformatCodeProcessor.lambda$prepareTask$2(ReformatCodeProcessor.java:133)
	at com.intellij.application.options.CodeStyle.doWithTemporarySettings(CodeStyle.java:345)
	at com.intellij.codeInsight.actions.ReformatCodeProcessor.lambda$prepareTask$3(ReformatCodeProcessor.java:129)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$ProcessingTask.lambda$performFileProcessing$7(AbstractLayoutCodeProcessor.java:474)
	at com.intellij.openapi.command.WriteCommandAction$BuilderImpl.lambda$doRunWriteCommandAction$1(WriteCommandAction.java:149)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:958)
	at com.intellij.openapi.command.WriteCommandAction$BuilderImpl.lambda$doRunWriteCommandAction$2(WriteCommandAction.java:147)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:219)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:184)
	at com.intellij.openapi.command.WriteCommandAction$BuilderImpl.doRunWriteCommandAction(WriteCommandAction.java:156)
	at com.intellij.openapi.command.WriteCommandAction$BuilderImpl.run(WriteCommandAction.java:123)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$ProcessingTask.lambda$performFileProcessing$8(AbstractLayoutCodeProcessor.java:474)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:209)
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:21)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:191)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:831)
	at com.intellij.openapi.application.impl.ApplicationImpl$3.run(ApplicationImpl.java:456)
	at com.intellij.openapi.application.impl.LaterInvocator$1.run(LaterInvocator.java:101)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:79)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:122)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:41)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:788)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:739)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:731)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:758)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:667)
	at com.intellij.ide.IdeEventQueue._dispatchEvent$lambda$7(IdeEventQueue.kt:571)
	at com.intellij.openapi.application.impl.ApplicationImpl.withoutImplicitRead(ApplicationImpl.java:1446)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:571)
	at com.intellij.ide.IdeEventQueue.access$_dispatchEvent(IdeEventQueue.kt:68)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:349)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:348)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:787)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:348)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:343)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$1(IdeEventQueue.kt:995)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:105)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:995)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$4(IdeEventQueue.kt:343)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:831)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:385)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
`

It is happening on MAC As well. To reproduce it we can create a Java file that has camel code and hit cmd+option+L to reformat the code, I think it should give below error.

   Got unexpected exception during formatting file:///test.java
   
   java.util.concurrent.ExecutionException: java.lang.IndexOutOfBoundsException: chars sequence.length:16162, start:1549, end:1542
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$ProcessingTask.checkStop(AbstractLayoutCodeProcessor.java:487)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$ProcessingTask.performFileProcessing(AbstractLayoutCodeProcessor.java:481)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor.lambda$runProcessFile$0(AbstractLayoutCodeProcessor.java:296)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$2.run(AbstractLayoutCodeProcessor.java:319)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:428)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:115)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressAsynchronously$6(CoreProgressManager.java:478)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:251)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:71)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:186)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:61)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:173)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:71)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$5(ProgressRunner.java:251)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$18(ProgressRunner.java:465)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IndexOutOfBoundsException: chars sequence.length:16162, start:1549, end:1542
	at com.intellij.util.text.CharSequenceSubSequence.<init>(CharSequenceSubSequence.java:33)
	at com.intellij.util.text.ImmutableText.subSequence(ImmutableText.java:206)
	at com.intellij.openapi.editor.impl.DocumentImpl$1.subSequence(DocumentImpl.java:95)
	at com.github.cameltooling.idea.service.extension.camel.JavaCamelIdeaUtils.formatText(JavaCamelIdeaUtils.java:479)
	at com.github.cameltooling.idea.service.extension.camel.JavaCamelIdeaUtils.lambda$format$4(JavaCamelIdeaUtils.java:442)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:992)
	at com.github.cameltooling.idea.service.extension.camel.JavaCamelIdeaUtils.format(JavaCamelIdeaUtils.java:442)
	at com.github.cameltooling.idea.service.extension.camel.JavaCamelIdeaUtils.processText(JavaCamelIdeaUtils.java:419)
	at com.github.cameltooling.idea.util.CamelIdeaUtils.processText(CamelIdeaUtils.java:285)
	at com.github.cameltooling.idea.formatter.CamelPostFormatProcessor.processText(CamelPostFormatProcessor.java:41)
	at com.intellij.psi.impl.source.codeStyle.CoreCodeStyleUtil.postProcessRange(CoreCodeStyleUtil.java:147)
	at com.intellij.psi.impl.source.codeStyle.CoreCodeStyleUtil.postProcessEnabledRanges(CoreCodeStyleUtil.java:138)
	at com.intellij.psi.impl.source.codeStyle.CoreCodeStyleUtil.postProcessText(CoreCodeStyleUtil.java:126)
	at com.intellij.formatting.service.CoreFormattingService.lambda$formatRanges$0(CoreFormattingService.java:64)
	at com.intellij.psi.impl.source.codeStyle.CoreCodeStyleUtil.postProcessRanges(CoreCodeStyleUtil.java:114)
	at com.intellij.formatting.service.CoreFormattingService.formatRanges(CoreFormattingService.java:64)
	at com.intellij.formatting.service.FormattingServiceUtil.formatRanges(FormattingServiceUtil.java:93)
	at com.intellij.psi.impl.source.codeStyle.CodeStyleManagerImpl.reformatText(CodeStyleManagerImpl.java:147)
	at com.intellij.psi.impl.source.codeStyle.CodeStyleManagerImpl.reformatText(CodeStyleManagerImpl.java:112)
	at com.intellij.codeInsight.actions.ReformatCodeProcessor.lambda$doReformat$6(ReformatCodeProcessor.java:201)
	at com.intellij.util.SlowOperations.allowSlowOperations(SlowOperations.java:196)
	at com.intellij.codeInsight.actions.ReformatCodeProcessor.lambda$doReformat$7(ReformatCodeProcessor.java:185)
	at com.intellij.openapi.editor.ex.util.EditorScrollingPositionKeeper.perform(EditorScrollingPositionKeeper.java:101)
	at com.intellij.codeInsight.actions.ReformatCodeProcessor.doReformat(ReformatCodeProcessor.java:185)
	at com.intellij.codeInsight.actions.ReformatCodeProcessor.lambda$prepareTask$2(ReformatCodeProcessor.java:133)
	at com.intellij.application.options.CodeStyle.doWithTemporarySettings(CodeStyle.java:344)
	at com.intellij.codeInsight.actions.ReformatCodeProcessor.lambda$prepareTask$3(ReformatCodeProcessor.java:129)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$ProcessingTask.lambda$performFileProcessing$7(AbstractLayoutCodeProcessor.java:478)
	at com.intellij.openapi.command.WriteCommandAction$BuilderImpl.lambda$doRunWriteCommandAction$1(WriteCommandAction.java:149)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:992)
	at com.intellij.openapi.command.WriteCommandAction$BuilderImpl.lambda$doRunWriteCommandAction$2(WriteCommandAction.java:147)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:219)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:184)
	at com.intellij.openapi.command.WriteCommandAction$BuilderImpl.doRunWriteCommandAction(WriteCommandAction.java:156)
	at com.intellij.openapi.command.WriteCommandAction$BuilderImpl.run(WriteCommandAction.java:123)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$ProcessingTask.lambda$performFileProcessing$8(AbstractLayoutCodeProcessor.java:478)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:208)
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:21)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:190)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:861)
	at com.intellij.openapi.application.impl.ApplicationImpl$4.run(ApplicationImpl.java:478)
	at com.intellij.openapi.application.impl.LaterInvocator$1.run(LaterInvocator.java:100)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:79)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:121)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:41)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:789)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:740)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:734)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:759)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:685)
	at com.intellij.ide.IdeEventQueue._dispatchEvent$lambda$10(IdeEventQueue.kt:589)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWithoutImplicitRead(ApplicationImpl.java:1485)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:589)
	at com.intellij.ide.IdeEventQueue.access$_dispatchEvent(IdeEventQueue.kt:67)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:369)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:368)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:787)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:368)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:363)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$1(IdeEventQueue.kt:992)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:105)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:992)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$7(IdeEventQueue.kt:363)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:861)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:405)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)

Sorry for the delay, check this test file

```
@Slf4j
public class Test extends RouteBuilder {
    @Override
    public void configure() {
        from("scheduler: test")
                .choice()
                .when(simple("condition1").isEqualTo("value1"))
                .process(exchange -> {
                    log.debug("execute process1");
                })
                .when(simple("condition2").isEqualTo("value2"))
                .process(exchange -> {
                    log.debug("execute process2");
                })
                .otherwise()
                .when(simple("condition3").isEqualTo("value3"))
                .to("direct:route3")
                .otherwise()
                .process(exchange -> {
                    log.debug("execute process4");
                })
                .filter(Exchange::isFailed)
                .to("direct:route4")
                .end();
        from("direct:route3")
                .process(exchange -> {
                    log.debug("execute route 3");
                });
        from("direct:route4")
                .process(exchange -> {
                    log.debug("execute route 4");
                });
    }
}
```

It will give errors like 

```
Got unexpected exception during formatting file:///Test.java
   
   java.util.concurrent.ExecutionException: java.lang.IndexOutOfBoundsException: chars sequence.length:1339, start:1308, end:2146
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$ProcessingTask.checkStop(AbstractLayoutCodeProcessor.java:487)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$ProcessingTask.performFileProcessing(AbstractLayoutCodeProcessor.java:481)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor.lambda$runProcessFile$0(AbstractLayoutCodeProcessor.java:296)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$2.run(AbstractLayoutCodeProcessor.java:319)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:428)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:115)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressAsynchronously$6(CoreProgressManager.java:478)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:251)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:71)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:186)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:61)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:173)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:71)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$5(ProgressRunner.java:251)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$18(ProgressRunner.java:465)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IndexOutOfBoundsException: chars sequence.length:1339, start:1308, end:2146
	at com.intellij.util.text.CharSequenceSubSequence.<init>(CharSequenceSubSequence.java:33)
	at com.intellij.util.text.ImmutableText.subSequence(ImmutableText.java:206)
	at com.intellij.openapi.editor.impl.DocumentImpl$1.subSequence(DocumentImpl.java:95)
	at com.github.cameltooling.idea.service.extension.camel.JavaCamelIdeaUtils.formatText(JavaCamelIdeaUtils.java:478)
	at com.github.cameltooling.idea.service.extension.camel.JavaCamelIdeaUtils.lambda$format$4(JavaCamelIdeaUtils.java:442)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:992)
	at com.github.cameltooling.idea.service.extension.camel.JavaCamelIdeaUtils.format(JavaCamelIdeaUtils.java:442)
	at com.github.cameltooling.idea.service.extension.camel.JavaCamelIdeaUtils.processText(JavaCamelIdeaUtils.java:419)
	at com.github.cameltooling.idea.util.CamelIdeaUtils.processText(CamelIdeaUtils.java:285)
	at com.github.cameltooling.idea.formatter.CamelPostFormatProcessor.processText(CamelPostFormatProcessor.java:41)
	at com.intellij.psi.impl.source.codeStyle.CoreCodeStyleUtil.postProcessRange(CoreCodeStyleUtil.java:147)
	at com.intellij.psi.impl.source.codeStyle.CoreCodeStyleUtil.postProcessEnabledRanges(CoreCodeStyleUtil.java:138)
	at com.intellij.psi.impl.source.codeStyle.CoreCodeStyleUtil.postProcessText(CoreCodeStyleUtil.java:126)
	at com.intellij.formatting.service.CoreFormattingService.lambda$formatRanges$0(CoreFormattingService.java:64)
	at com.intellij.psi.impl.source.codeStyle.CoreCodeStyleUtil.postProcessRanges(CoreCodeStyleUtil.java:114)
	at com.intellij.formatting.service.CoreFormattingService.formatRanges(CoreFormattingService.java:64)
	at com.intellij.formatting.service.FormattingServiceUtil.formatRanges(FormattingServiceUtil.java:93)
	at com.intellij.psi.impl.source.codeStyle.CodeStyleManagerImpl.reformatText(CodeStyleManagerImpl.java:147)
	at com.intellij.psi.impl.source.codeStyle.CodeStyleManagerImpl.reformatText(CodeStyleManagerImpl.java:112)
	at com.intellij.codeInsight.actions.ReformatCodeProcessor.lambda$doReformat$6(ReformatCodeProcessor.java:201)
	at com.intellij.util.SlowOperations.allowSlowOperations(SlowOperations.java:196)
	at com.intellij.codeInsight.actions.ReformatCodeProcessor.lambda$doReformat$7(ReformatCodeProcessor.java:185)
	at com.intellij.openapi.editor.ex.util.EditorScrollingPositionKeeper.perform(EditorScrollingPositionKeeper.java:101)
	at com.intellij.codeInsight.actions.ReformatCodeProcessor.doReformat(ReformatCodeProcessor.java:185)
	at com.intellij.codeInsight.actions.ReformatCodeProcessor.lambda$prepareTask$2(ReformatCodeProcessor.java:133)
	at com.intellij.application.options.CodeStyle.doWithTemporarySettings(CodeStyle.java:367)
	at com.intellij.codeInsight.actions.ReformatCodeProcessor.lambda$prepareTask$3(ReformatCodeProcessor.java:129)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$ProcessingTask.lambda$performFileProcessing$7(AbstractLayoutCodeProcessor.java:478)
	at com.intellij.openapi.command.WriteCommandAction$BuilderImpl.lambda$doRunWriteCommandAction$1(WriteCommandAction.java:149)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:992)
	at com.intellij.openapi.command.WriteCommandAction$BuilderImpl.lambda$doRunWriteCommandAction$2(WriteCommandAction.java:147)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:219)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:184)
	at com.intellij.openapi.command.WriteCommandAction$BuilderImpl.doRunWriteCommandAction(WriteCommandAction.java:156)
	at com.intellij.openapi.command.WriteCommandAction$BuilderImpl.run(WriteCommandAction.java:123)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor$ProcessingTask.lambda$performFileProcessing$8(AbstractLayoutCodeProcessor.java:478)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:208)
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:21)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:190)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:861)
	at com.intellij.openapi.application.impl.ApplicationImpl$4.run(ApplicationImpl.java:478)
	at com.intellij.openapi.application.impl.LaterInvocator$1.run(LaterInvocator.java:100)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:79)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:121)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:41)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:792)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:739)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:733)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:761)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:685)
	at com.intellij.ide.IdeEventQueue._dispatchEvent$lambda$10(IdeEventQueue.kt:589)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWithoutImplicitRead(ApplicationImpl.java:1485)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:589)
	at com.intellij.ide.IdeEventQueue.access$_dispatchEvent(IdeEventQueue.kt:67)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:369)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:368)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:787)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:368)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:363)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$1(IdeEventQueue.kt:992)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:105)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:992)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$7(IdeEventQueue.kt:363)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:861)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:405)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
```

--- a/camel-idea-plugin/src/main/java/com/github/cameltooling/idea/service/extension/camel/JavaCamelIdeaUtils.java
+++ b/camel-idea-plugin/src/main/java/com/github/cameltooling/idea/service/extension/camel/JavaCamelIdeaUtils.java
@@ -50,6 +50,7 @@ import com.intellij.psi.PsiMethod;
 import com.intellij.psi.PsiMethodCallExpression;
 import com.intellij.psi.PsiReferenceExpression;
 import com.intellij.psi.PsiType;
+import com.intellij.psi.PsiTypeParameter;
 import com.intellij.psi.codeStyle.CodeStyleSettings;
 import com.intellij.psi.impl.source.PostprocessReformattingAspect;
 import com.intellij.psi.impl.source.PsiClassReferenceType;
@@ -63,6 +64,7 @@ import java.util.ArrayDeque;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
+import java.util.Comparator;
 import java.util.Deque;
 import java.util.List;
 import java.util.Objects;
@@ -460,7 +462,11 @@ public class JavaCamelIdeaUtils extends CamelIdeaUtils implements CamelIdeaUtils
     private void formatText(PsiFile file, Document document, int startOffset, int endOffset,
                             CodeStyleSettings settings) {
         Module module = ModuleUtilCore.findModuleForPsiElement(file.getOriginalElement());
-        List<PsiElement> endpointDeclarations = findEndpointDeclarations(module, e -> true);
+        List<PsiElement> endpointDeclarations = findEndpointDeclarations(module, e -> true)
+            .stream()
+            .filter(e -> file.equals(e.getContainingFile()))
+            .sorted(Comparator.comparingInt(PsiElement::getTextOffset))
+            .toList();
         Deque<Consumer<Document>> changes = new ArrayDeque<>();
         boolean useTabCharacter = settings.useTabCharacter(JavaFileType.INSTANCE);
         int indentSize = settings.getIndentSize(JavaFileType.INSTANCE);
@@ -475,11 +481,13 @@ public class JavaCamelIdeaUtils extends CamelIdeaUtils implements CamelIdeaUtils
 
                 TextRange textRange = parent.getTextRange();
                 CharSequence charsSequence = document.getCharsSequence();
-                CharSequence contentToFormat = charsSequence.subSequence(textRange.getStartOffset(), textRange.getEndOffset());
-                CharSequence linePrefix = charsSequence.subSequence(document.getLineStartOffset(document.getLineNumber(textOffset)), textRange.getStartOffset());
-                CharSequence result = formatText(file, textRange.getStartOffset(), endOffset, contentToFormat, linePrefix, useTabCharacter, indentSize);
+                int textRangeStartOffset = textRange.getStartOffset();
+                int textRangeEndOffset = textRange.getEndOffset();
+                CharSequence contentToFormat = charsSequence.subSequence(textRangeStartOffset, textRangeEndOffset);
+                CharSequence linePrefix = charsSequence.subSequence(document.getLineStartOffset(document.getLineNumber(textOffset)), textRangeStartOffset);
+                CharSequence result = formatText(file, textRangeStartOffset, endOffset, contentToFormat, linePrefix, useTabCharacter, indentSize);
                 // Ensure to apply the last changes first to avoid having to deal with offset change
-                changes.addFirst(doc -> doc.replaceString(textRange.getStartOffset(), textRange.getEndOffset(), result));
+                changes.addFirst(doc -> doc.replaceString(textRangeStartOffset, textRangeEndOffset, result));
             }
         }
         Consumer<Document> change = changes.poll();
@@ -525,7 +533,8 @@ public class JavaCamelIdeaUtils extends CamelIdeaUtils implements CamelIdeaUtils
                 if (!hasDot && SUB_DSL_ROOTS.contains(methodName)) {
                     stack.push(methodName);
                     indent++;
-                    if (Objects.nonNull(previousMethod) && !ADD_INDENT.contains(previousMethod) && !SUB_DSL_ROOTS.contains(previousMethod)) {
+                    if (Objects.nonNull(previousMethod) && !ADD_INDENT.contains(previousMethod) && !SUB_DSL_ROOTS.contains(previousMethod)
+                        && !ADD_INDENT_OR_NEW_BLOCK.contains(previousMethod)) {
                         indent++;
                         currentIndent++;
                     }
@@ -560,7 +569,8 @@ public class JavaCamelIdeaUtils extends CamelIdeaUtils implements CamelIdeaUtils
                         currentIndent--;
                     }
                 }
-                if (currentIndent > 0) {
+                boolean isShortPredicate = isShortPredicate(file, offset, result);
+                if (currentIndent > 0 && !isShortPredicate) {
                     appendNewLine(result, linePrefix, useTabCharacter, indentSize, currentIndent);
                 }
                 if (hasDot) {
@@ -570,7 +580,7 @@ public class JavaCamelIdeaUtils extends CamelIdeaUtils implements CamelIdeaUtils
                 result.append('(');
                 lastIndex = matcher.end();
                 previousMethod = methodName;
-                lastProcessedMethodIsFormatted = true;
+                lastProcessedMethodIsFormatted = !isShortPredicate && !isPredicateOfWhenClause(file, offset);
                 continue;
             }
             lastProcessedMethodIsFormatted = false;
@@ -611,6 +621,19 @@ public class JavaCamelIdeaUtils extends CamelIdeaUtils implements CamelIdeaUtils
      * @return {@code true} if it is method to format, {@code false} otherwise.
      */
     private static boolean isMethodToFormat(PsiFile file, int offset) {
+        return testMethod(file, offset, JavaCamelIdeaUtils::isMethodToFormat);
+    }
+
+    /**
+     * Indicates whether the {@code PsiElement} located at the given {@code offset} is a method that matches with the
+     * given predicate.
+     *
+     * @param file   the file in which the method is located.
+     * @param offset the offset of the method to check.
+     * @param predicate the predicate to evaluate
+     * @return {@code true} if it is method matches with the predicate, {@code false} otherwise.
+     */
+    private static boolean testMethod(PsiFile file, int offset, Predicate<? super PsiMethod> predicate) {
         PsiElement element = file.findElementAt(offset);
         if (element == null) {
             LOG.debug("No element cannot be found at index %d".formatted(offset));
@@ -625,14 +648,70 @@ public class JavaCamelIdeaUtils extends CamelIdeaUtils implements CamelIdeaUtils
                 } else if (Arrays.stream(results).map(JavaResolveResult::getElement)
                     .filter(PsiMethod.class::isInstance)
                     .map(PsiMethod.class::cast)
-                    .anyMatch(JavaCamelIdeaUtils::isMethodToFormat)) {
+                    .anyMatch(predicate)) {
                     return true;
                 } else {
-                    LOG.trace("The method is not part of the methods to format");
+                    LOG.trace("The method doesn't match with the predicate");
                 }
             }
         } else {
             LOG.trace("The element at index %d is not a PsiIdentifier".formatted(offset));
+        }
+        return false;
+    }
+
+    /**
+     * Indicates whether the {@code PsiElement} located at the given {@code offset} is a predicate that can be inlined.
+     *
+     * @param file   the file in which the method is located.
+     * @param offset the offset of the method to check.
+     * @param currentContent the current formatted content
+     * @return {@code true} if it is a predicate that can be inlined, {@code false} otherwise.
+     */
+    private static boolean isShortPredicate(PsiFile file, int offset, StringBuilder currentContent) {
+        return checkPredicate(file, offset, methodName -> currentContent.lastIndexOf(methodName) > currentContent.lastIndexOf("\n"));
+    }
+
+    /**
+     * Indicates whether the {@code PsiElement} located at the given {@code offset} is a predicate of a where clause.
+     *
+     * @param file   the file in which the method is located.
+     * @param offset the offset of the method to check.
+     * @return {@code true} if it is a predicate of a where clause, {@code false} otherwise.
+     */
+    private static boolean isPredicateOfWhenClause(PsiFile file, int offset) {
+        return checkPredicate(file, offset, "when"::equals);
+    }
+
+    /**
+     * Indicates whether the {@code PsiElement} located at the given {@code offset} is a Camel predicate that matches with the
+     * given predicate.
+     *
+     * @param file   the file in which the method is located.
+     * @param offset the offset of the method to check.
+     * @param predicate the predicate to evaluate
+     * @return {@code true} if it is method matches with the predicate, {@code false} otherwise.
+     */
+    private static boolean checkPredicate(PsiFile file, int offset, Predicate<String> predicate) {
+        if (testMethod(file, offset, JavaCamelIdeaUtils::isPredicate)) {
+            PsiExpressionList parent = PsiTreeUtil.getParentOfType(file.findElementAt(offset), PsiExpressionList.class);
+            if (parent == null) {
+                LOG.debug("The parent PsiExpressionList of the element at index %d cannot be found".formatted(offset));
+            } else {
+                PsiMethodCallExpression parentMethodCallExpression = PsiTreeUtil.getParentOfType(parent.getPrevSibling(), PsiMethodCallExpression.class);
+                if (parentMethodCallExpression == null) {
+                    LOG.debug("The parent method corresponding to the element at index %d cannot be resolved".formatted(offset));
+                } else {
+                    String methodName = parentMethodCallExpression.getMethodExpression().getReferenceName();
+                    if (methodName == null) {
+                        LOG.debug("The name of the parent method corresponding to the element at index %d cannot be found".formatted(offset));
+                    } else {
+                        return predicate.test(methodName);
+                    }
+                }
+            }
+        } else {
+            LOG.trace("The element at index %d is not a Predicate".formatted(offset));
         }
         return false;
     }
@@ -695,6 +774,16 @@ public class JavaCamelIdeaUtils extends CamelIdeaUtils implements CamelIdeaUtils
     }
 
     /**
+     * Indicates whether the given method is a predicate
+     *
+     * @param method the method to check
+     * @return {@code true} if the method is a predicate, {@code false} otherwise.
+     */
+    private static boolean isPredicate(PsiMethod method) {
+        return "org.apache.camel.Predicate".equals(getReturnType(method));
+    }
+
+    /**
      * Gives the containing class of the given method.
      *
      * @param method the method for which the containing class is expected
@@ -721,6 +810,13 @@ public class JavaCamelIdeaUtils extends CamelIdeaUtils implements CamelIdeaUtils
             PsiClass returnType = referenceType.resolve();
             if (returnType == null) {
                 LOG.trace("The return type of the method cannot be resolved");
+            } else if (returnType instanceof PsiTypeParameter type) {
+                PsiClass superClass = type.getSuperClass();
+                if (superClass == null) {
+                    LOG.trace("The super class of the type parameter of the method cannot be resolved");
+                } else {
+                    return superClass.getQualifiedName();
+                }
             } else {
                 return returnType.getQualifiedName();
             }

--- a/camel-idea-plugin/src/test/java/com/github/cameltooling/idea/formatter/CamelPostFormatProcessorIT.java
+++ b/camel-idea-plugin/src/test/java/com/github/cameltooling/idea/formatter/CamelPostFormatProcessorIT.java
@@ -34,6 +34,7 @@ import org.jetbrains.annotations.Nullable;
 public class CamelPostFormatProcessorIT extends CamelLightCodeInsightFixtureTestCaseIT {
 
     private static final String CAMEL_SUPPORT_MAVEN_ARTIFACT = String.format("org.apache.camel:camel-support:%s", CAMEL_VERSION);
+    private static final String CAMEL_API_MAVEN_ARTIFACT = String.format("org.apache.camel:camel-api:%s", CAMEL_VERSION);
 
     /**
      * Ensures that expressions are not formatted and the language DSL is supported.
@@ -64,6 +65,14 @@ public class CamelPostFormatProcessorIT extends CamelLightCodeInsightFixtureTest
     }
 
     /**
+     * Ensures that the test case defined in <a href="https://github.com/camel-tooling/camel-idea-plugin/issues/879">#879</a>
+     * is properly fixed.
+     */
+    public void testFormatting879() {
+        doTest("Formatting879", null);
+    }
+
+    /**
      * Ensures that a partial format not including a route has no effect.
      */
     public void testPartialFormat() {
@@ -80,7 +89,7 @@ public class CamelPostFormatProcessorIT extends CamelLightCodeInsightFixtureTest
     @Nullable
     @Override
     protected String[] getMavenDependencies() {
-        return new String[]{CAMEL_CORE_MODEL_MAVEN_ARTIFACT, CAMEL_SUPPORT_MAVEN_ARTIFACT};
+        return new String[]{CAMEL_CORE_MODEL_MAVEN_ARTIFACT, CAMEL_SUPPORT_MAVEN_ARTIFACT, CAMEL_API_MAVEN_ARTIFACT};
     }
 
     @Override
@@ -92,19 +101,23 @@ public class CamelPostFormatProcessorIT extends CamelLightCodeInsightFixtureTest
         var before = "before/%s.java".formatted(name);
         var after = "after/%s.txt".formatted(name);
         myFixture.configureByFile(before);
-        if (range == null) {
-            range = getFile().getTextRange();
-        }
-        performReformatting(getProject(), getFile(), range);
+        performReformatting(range);
         myFixture.checkResultByFile(after);
         //check idempotence of formatter
-        performReformatting(getProject(), getFile(), range);
+        performReformatting(range);
         myFixture.checkResultByFile(after);
     }
 
-    private void performReformatting(Project project, PsiFile file, TextRange range) {
+    private void performReformatting(TextRange range) {
+        Project project = getProject();
         WriteCommandAction.runWriteCommandAction(
-            project, () -> CodeStyleManager.getInstance(project).reformatText(file, List.of(range))
+            project, () -> {
+                TextRange textRange = range;
+                if (textRange == null) {
+                    textRange = getFile().getTextRange();
+                }
+                CodeStyleManager.getInstance(project).reformatText(getFile(), List.of(textRange));
+            }
         );
         PlatformTestUtil.dispatchAllInvocationEventsInIdeEventQueue();
     }

--- a/camel-idea-plugin/src/test/resources/testData/formatter/after/Formatting879.txt
+++ b/camel-idea-plugin/src/test/resources/testData/formatter/after/Formatting879.txt
@@ -1,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import org.apache.camel.Exchange;
+import org.apache.camel.builder.RouteBuilder;
+
+public class Formatting879 extends RouteBuilder {
+    @Override
+    public void configure() {
+        from("scheduler: test")
+            .choice()
+                .when(simple("condition1").isEqualTo("value1"))
+                    .process(exchange -> log.debug("execute process1"))
+                    .to("direct:route1")
+                .when(simple("condition2").isEqualTo("value2"))
+                    .process(exchange -> log.debug("execute process2"))
+                .otherwise()
+                    .when(simple("condition3").isEqualTo("value3"))
+                        .to("direct:route3")
+                    .otherwise()
+                        .process(exchange -> log.debug("execute process4"))
+                        .filter(Exchange::isFailed)
+                        .to("direct:route4")
+                    .end();
+        from("direct:route3")
+            .process(exchange -> log.debug("execute route 3"));
+        from("direct:route4")
+            .process(exchange -> log.debug("execute route 4"));
+    }
+}

--- a/camel-idea-plugin/src/test/resources/testData/formatter/after/RouteWithExpressions.txt
+++ b/camel-idea-plugin/src/test/resources/testData/formatter/after/RouteWithExpressions.txt
@@ -65,6 +65,18 @@ public class RouteWithExpressions extends RouteBuilder {
                     .end())
                 .isEqualTo("bar"))
             .to("mock:d");
+        from("direct:e")
+            .choice()
+                .when(
+                    expression(
+                        expression()
+                            .header()
+                            .expression("foo")
+                        .end())
+                    .isEqualTo("bar"))
+                    .to("mock:e1")
+                .otherwise()
+                    .to("mock:e2");
 
         AdviceWith.adviceWith(context, "advice-with-on-exception-transacted-test-route", a -> {
             a.weaveAddFirst().transform(constant("Bye World"));

--- a/camel-idea-plugin/src/test/resources/testData/formatter/before/Formatting879.java
+++ b/camel-idea-plugin/src/test/resources/testData/formatter/before/Formatting879.java
@@ -1,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import org.apache.camel.Exchange;
+import org.apache.camel.builder.RouteBuilder;
+
+public class Formatting879 extends RouteBuilder {
+    @Override
+    public void configure() {
+        from("scheduler: test")
+            .choice()
+            .when(simple("condition1").isEqualTo("value1"))
+            .process(exchange -> log.debug("execute process1"))
+            .to("direct:route1")
+            .when(simple("condition2").isEqualTo("value2"))
+            .process(exchange -> log.debug("execute process2"))
+            .otherwise()
+            .when(simple("condition3").isEqualTo("value3"))
+            .to("direct:route3")
+            .otherwise()
+            .process(exchange -> log.debug("execute process4"))
+            .filter(Exchange::isFailed)
+            .to("direct:route4")
+            .end();
+        from("direct:route3")
+            .process(exchange -> log.debug("execute route 3"));
+        from("direct:route4")
+            .process(exchange -> log.debug("execute route 4"));
+    }
+}

--- a/camel-idea-plugin/src/test/resources/testData/formatter/before/RouteWithExpressions.java
+++ b/camel-idea-plugin/src/test/resources/testData/formatter/before/RouteWithExpressions.java
@@ -42,6 +42,7 @@ public class RouteWithExpressions extends RouteBuilder {
         from("direct:b").setBody().expression(expression().simple().expression("Hello World Out").end()).to("mock:b");
         from("direct:c").setBody(expression().simple().expression("Hello World In").end()).to("mock:c");
         from("direct:d").filter(expression(expression().header().expression("foo").end()).isEqualTo("bar")).to("mock:d");
+        from("direct:e").choice().when(expression(expression().header().expression("foo").end()).isEqualTo("bar")).to("mock:e1").otherwise().to("mock:e2");
 
         AdviceWith.adviceWith(context, "advice-with-on-exception-transacted-test-route", a -> {
             a.weaveAddFirst().transform(constant("Bye World"));

